### PR TITLE
chore(Input | Checkbox): remove hard icon dependencies

### PR DESCRIPTION
**Pull request by mnajdova**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/12183)

### CODE REVIEW

File: packages/fluentui/react/src/components/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -156,10 +156,9 @@ const Checkbox: React.FC<WithAsProp<CheckboxProps>> &
       {Icon.create(icon, {
         defaultProps: () =>
           getA11Props('icon', {
```
Comment (pompomon) [inline packages/fluentui/react/src/components/Checkbox/Checkbox.tsx]:
```suggestion
          getA11yProps('icon', {
```

File: packages/fluentui/react/src/components/Input/Input.tsx [hunk 2]
```diff
@@ -202,7 +202,7 @@ class Input extends AutoControlledComponent<WithAsProp<InputProps>, InputState>
     const { value } = this.state;
 
     if (clearable && (value as string).length !== 0) {
```
Comment (layershifter) [inline packages/fluentui/react/src/components/Input/Input.tsx]: Now it should be something like:

```
if (clearable && (value as string).length !== 0) {
  return <Box styles />
}

return Icon.create(icon)
```

File: packages/fluentui/react/src/components/Checkbox/Checkbox.tsx [hunk 3]
```diff
@@ -156,10 +156,9 @@ const Checkbox: React.FC<WithAsProp<CheckboxProps>> &
       {Icon.create(icon, {
```
Comment (layershifter) [inline packages/fluentui/react/src/components/Checkbox/Checkbox.tsx]: Box?

File: packages/fluentui/react/src/components/Checkbox/Checkbox.tsx [hunk 4]
```diff
@@ -156,10 +156,9 @@ const Checkbox: React.FC<WithAsProp<CheckboxProps>> &
...
         defaultProps: () =>
           getA11Props('icon', {
-            outline: toggle && !state.checked,
             size: toggle ? 'medium' : 'smaller',
             className: Checkbox.slotClassNames.indicator,
-            name: toggle ? 'icon-circle' : 'icon-checkmark',
+            name: '',
```
[the remark this answers is not in the document]
Comment (mnajdova) [reply packages/fluentui/react/src/components/Checkbox/Checkbox.tsx]: name is required prop

File: packages/fluentui/react/src/components/Checkbox/Checkbox.tsx (hunk 3 above)
Comment (mnajdova) [reply packages/fluentui/react/src/components/Checkbox/Checkbox.tsx]: Not sure if Box is valid here, as from client's perspective I think icon is the valid slot

File: packages/fluentui/react/src/components/Input/Input.tsx (hunk 2 above)
Comment (mnajdova) [reply packages/fluentui/react/src/components/Input/Input.tsx]: We cannot have children in the Icon component

File: packages/fluentui/react/src/components/Checkbox/Checkbox.tsx (hunk 3 above)
Comment (layershifter) [reply packages/fluentui/react/src/components/Checkbox/Checkbox.tsx]: I don't see any reasons to be honest 🤔  You will not use `book` for as checkmark

File: packages/fluentui/react/src/components/Input/Input.tsx (hunk 2 above)
Comment (layershifter) [reply packages/fluentui/react/src/components/Input/Input.tsx]: I see what you mean, but it's really weird

File: packages/fluentui/react/src/themes/teams/components/Checkbox/checkboxStyles.ts [hunk 5]
```diff
@@ -80,53 +116,61 @@ const checkboxStyles: ComponentSlotStylesPrepared<CheckboxStylesProps, CheckboxV
...
     ...(p.disabled && {
-      color: v.disabledToggleIndicatorColor,
-      background: v.disabledBackground,
-      borderColor: v.disabledBorderColor
-    }),
-
-    ...(p.disabled &&
-      p.checked && {
-        color: v.disabledCheckedIndicatorColor,
+      ...(!p.checked && {
```
Comment (layershifter) [inline packages/fluentui/react/src/themes/teams/components/Checkbox/checkboxStyles.ts]: I don't understand this... Why `!p.checked` is required there? Should it be just:

```
...(p.disabled && {
  background: v.disabledBackground,
  borderColor: v.disabledBorderColor
  ...(p.checked && {
    background: v.disabledBackgroundChecked,
    borderColor: 'transparent',
  }),
}
```

File: packages/fluentui/react/src/themes/teams/components/Checkbox/checkboxStyles.ts [hunk 6]
```diff
@@ -80,53 +116,61 @@ const checkboxStyles: ComponentSlotStylesPrepared<CheckboxStylesProps, CheckboxV
...
     },
 
-    ...(p.checked && {
-      background: v.checkedBackground,
-      borderColor: v.checkedBorderColor,
-      color: v.checkedIndicatorColor,
-      padding: v.toggleCheckedPadding
-    }),
+    ...(p.checked &&
+      !p.disabled && {
```
Comment (layershifter) [inline packages/fluentui/react/src/themes/teams/components/Checkbox/checkboxStyles.ts]: https://github.com/OfficeDev/office-ui-fabric-react/pull/12183/files#r388786483

Same thing about `!p.disabled`... We have following states:
- default
- `checked`
- `disabled`
- `checked` & `disabled`

Why we need negative conditions for other states here?

File: packages/fluentui/react/src/themes/teams/components/Input/inputStyles.ts [hunk 7]
```diff
@@ -49,18 +51,31 @@ const inputStyles: ComponentSlotStylesPrepared<InputProps, InputVariables> = {
     })
   }),
 
-  icon: ({ props: p, variables: v }): ICSSInJSStyle => ({
-    color: v.iconColor,
-    outline: 0,
-    position: v.iconPosition as PositionProperty,
+  icon: ({ props: p, variables: v }): ICSSInJSStyle => {
+    const hasValue = !!p.value && (p.value as string).length !== 0;
```
Comment (layershifter) [inline packages/fluentui/react/src/themes/teams/components/Input/inputStyles.ts]: Can we move thing to `getAutoControlledStateFromProps()` for now? And then we can just reuse it in `useStyles`

**Pull request merged**